packages/query-core/src/types.ts

```diff
@@ -732,6 +733,7 @@ export interface MutationObserverBaseResult<
   isSuccess: boolean
   mutate: MutateFunction<TData, TError, TVariables, TContext>
   reset: () => void
+  status: MutationStatus
```

Comment:
![스크린샷 2023-11-05 오전 3 44 05](https://github.com/TanStack/query/assets/64779472/c0a2da0e-f5a2-4e40-b99d-e234fb9f5e55)

I think this is safer and clearer because if you actually put in an incompatible state, it will throw an error.

Reply:
The other tasks have similar intentions.
They're meant to make your code more consistent, and they can be clearer and safer.